### PR TITLE
fix: excepting import error

### PR DESCRIPTION
**Pull request by Anokhi1994**
We get an error 
    from conan.tools.build import cross_building
ImportError: cannot import name 'cross_building'

and to fix this , we must except the error instead of crashing the program. 

Hope it is okay. 

---

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libcurl/all/conanfile.py [hunk 1]
```diff
@@ -5,7 +5,10 @@
 from conans import ConanFile, AutoToolsBuildEnvironment, CMake, tools
 from conans.errors import ConanInvalidConfiguration
 from conan.tools.microsoft import is_msvc
-from conan.tools.build import cross_building
+try:
+    from conan.tools.cross_building import cross_building
+except ImportError:
+    from conan.tools.build.cross_building import cross_building
```
Comment (uilianries) [inline recipes/libcurl/all/conanfile.py:11]:
```suggestion
from conan.tools.build.cross_building import cross_building
```
We need to move forward to Conan 2.0

**Review by uilianries — CHANGES_REQUESTED** [review]

**Review by ericLemanissier — APPROVED** [review]

File: recipes/libcurl/all/conanfile.py [hunk 2]
```diff
@@ -5,7 +5,7 @@
 from conans import ConanFile, AutoToolsBuildEnvironment, CMake, tools
 from conans.errors import ConanInvalidConfiguration
 from conan.tools.microsoft import is_msvc
-from conan.tools.build import cross_building
+from conan.tools.build.cross_building import cross_building
 
 required_conan_version = ">=1.45.0"
```
Comment (prince-chrismc) [inline recipes/libcurl/all/conanfile.py:10]: Should the version be bumped? I image there a release where that was changed

Comment (uilianries) [reply recipes/libcurl/all/conanfile.py:10]: Yes, please.

Comment (Anokhi1994) [reply recipes/libcurl/all/conanfile.py]: I did this change, but now other things break..

**Comment by prince-chrismc** [comment]
> Module 'conans.tools' has no 'Version' member.

Bah! new linter rules are flagging error, upgrade docs are here
https://github.com/conan-io/conan-center-index/blob/master/docs/v2_linter.md#import-tools-from-conan

File: recipes/libcurl/all/conanfile.py [hunk 3]
```diff
@@ -5,9 +5,9 @@
 from conans import ConanFile, AutoToolsBuildEnvironment, CMake, tools
 from conans.errors import ConanInvalidConfiguration
```
Comment (Anokhi1994) [inline recipes/libcurl/all/conanfile.py:6]:
```suggestion
from conan import ConanFile, AutoToolsBuildEnvironment, CMake, tools
from conan.errors import ConanInvalidConfiguration
```

**Comment by Anokhi1994** [comment]
It looks conan.tools has no members that existed before as per Linter.  Not sure what to do here.

File: recipes/libcurl/all/conanfile.py [hunk 4]
```diff
@@ -2,12 +2,12 @@
 import os
 import re
 import functools
-from conans import ConanFile, AutoToolsBuildEnvironment, CMake, tools
+from conan import ConanFile, AutoToolsBuildEnvironment, CMake, tools
```
Comment (ericLemanissier) [inline recipes/libcurl/all/conanfile.py:5]:
```suggestion
from conan import ConanFile
from conans import AutoToolsBuildEnvironment, CMake, tools
```

File: recipes/libcurl/all/conanfile.py [hunk 5]
```diff
@@ -136,12 +136,12 @@ def _is_using_cmake_build(self):
 
     @property
     def _has_zstd_option(self):
-        return tools.Version(self.version) >= "7.72.0"
+        return tools.scm.Version(self.version) >= "7.72.0"
```
Comment (ericLemanissier) [inline recipes/libcurl/all/conanfile.py:139]:
```suggestion
        return Version(self.version) >= "7.72.0"
```

File: recipes/libcurl/all/conanfile.py [hunk 6]
```diff
@@ -136,12 +136,12 @@ def _is_using_cmake_build(self):
...
 
     @property
     def _has_metalink_option(self):
         # Support for metalink was removed in version 7.78.0 https://github.com/curl/curl/pull/7176
-        return tools.Version(self.version) < "7.78.0" and not self._is_using_cmake_build
+        return tools.scm.Version(self.version) < "7.78.0" and not self._is_using_cmake_build
```
Comment (ericLemanissier) [inline recipes/libcurl/all/conanfile.py:144]:
```suggestion
        return Version(self.version) < "7.78.0" and not self._is_using_cmake_build
```

File: recipes/libcurl/all/conanfile.py [hunk 7]
```diff
@@ -167,7 +167,7 @@ def configure(self):
 
         # These options are not used in CMake build yet
         if self._is_using_cmake_build:
-            if tools.Version(self.version) < "7.75.0":
+            if tools.scm.Version(self.version) < "7.75.0":
```
Comment (ericLemanissier) [inline recipes/libcurl/all/conanfile.py:170]:
```suggestion
            if Version(self.version) < "7.75.0":
```

File: recipes/libcurl/all/conanfile.py [hunk 8]
```diff
@@ -194,7 +194,7 @@ def validate(self):
             raise ConanInvalidConfiguration("schannel only suppported on Windows.")
         if self.options.with_ssl == "darwinssl" and not tools.is_apple_os(self.settings.os):
             raise ConanInvalidConfiguration("darwinssl only suppported on Apple like OS (Macos, iOS, watchOS or tvOS).")
-        if self.options.with_ssl == "wolfssl" and self._is_using_cmake_build and tools.Version(self.version) < "7.70.0":
+        if self.options.with_ssl == "wolfssl" and self._is_using_cmake_build and tools.scm.Version(self.version) < "7.70.0":
```
Comment (ericLemanissier) [inline recipes/libcurl/all/conanfile.py:197]:
```suggestion
        if self.options.with_ssl == "wolfssl" and self._is_using_cmake_build and Version(self.version) < "7.70.0":
```

File: recipes/libcurl/all/conanfile.py [hunk 9]
```diff
@@ -327,7 +327,7 @@ def _patch_cmake(self):
 
         # INTERFACE_LIBRARY (generated by the cmake_find_package generator) targets doesn't have the LOCATION property.
         # So skipp the LOCATION check in the CMakeLists.txt
-        if tools.Version(self.version) >= "7.80.0":
+        if tools.scm.Version(self.version) >= "7.80.0":
```
Comment (ericLemanissier) [inline recipes/libcurl/all/conanfile.py:330]:
```suggestion
        if Version(self.version) >= "7.80.0":
```

File: recipes/libcurl/all/conanfile.py [hunk 10]
```diff
@@ -415,7 +415,7 @@ def _get_configure_command_args(self):
 
         # ntlm will default to enabled if any SSL options are enabled
         if not self.options.with_ntlm:
-            if tools.Version(self.version) <= "7.77.0":
+            if tools.scm.Version(self.version) <= "7.77.0":
```
Comment (ericLemanissier) [inline recipes/libcurl/all/conanfile.py:418]:
```suggestion
            if Version(self.version) <= "7.77.0":
```

File: recipes/libcurl/all/conanfile.py [hunk 11]
```diff
@@ -532,26 +532,26 @@ def _configure_cmake(self):
         cmake.definitions["BUILD_SHARED_LIBS"] = self.options.shared
         cmake.definitions["CURL_STATICLIB"] = not self.options.shared
         cmake.definitions["CMAKE_DEBUG_POSTFIX"] = ""
-        if tools.Version(self.version) >= "7.81.0":
+        if tools.scm.Version(self.version) >= "7.81.0":
```
Comment (ericLemanissier) [inline recipes/libcurl/all/conanfile.py:535]:
```suggestion
        if Version(self.version) >= "7.81.0":
```

File: recipes/libcurl/all/conanfile.py [hunk 12]
```diff
@@ -532,26 +532,26 @@ def _configure_cmake(self):
...
             cmake.definitions["CURL_USE_SCHANNEL"] = self.options.with_ssl == "schannel"
-        elif tools.Version(self.version) >= "7.72.0":
+        elif tools.scm.Version(self.version) >= "7.72.0":
```
Comment (ericLemanissier) [inline recipes/libcurl/all/conanfile.py:537]:
```suggestion
        elif Version(self.version) >= "7.72.0":
```

File: recipes/libcurl/all/conanfile.py [hunk 13]
```diff
@@ -532,26 +532,26 @@ def _configure_cmake(self):
...
             cmake.definitions["CMAKE_USE_SCHANNEL"] = self.options.with_ssl == "schannel"
         else:
             cmake.definitions["CMAKE_USE_WINSSL"] = self.options.with_ssl == "schannel"
-        if tools.Version(self.version) >= "7.81.0":
+        if tools.scm.Version(self.version) >= "7.81.0":
```
Comment (ericLemanissier) [inline recipes/libcurl/all/conanfile.py:541]:
```suggestion
        if Version(self.version) >= "7.81.0":
```

File: recipes/libcurl/all/conanfile.py [hunk 14]
```diff
@@ -532,26 +532,26 @@ def _configure_cmake(self):
...
             cmake.definitions["CURL_USE_OPENSSL"] = self.options.with_ssl == "openssl"
         else:
             cmake.definitions["CMAKE_USE_OPENSSL"] = self.options.with_ssl == "openssl"
-        if tools.Version(self.version) >= "7.81.0":
+        if tools.scm.Version(self.version) >= "7.81.0":
```
Comment (ericLemanissier) [inline recipes/libcurl/all/conanfile.py:545]:
```suggestion
        if Version(self.version) >= "7.81.0":
```

File: recipes/libcurl/all/conanfile.py [hunk 15]
```diff
@@ -532,26 +532,26 @@ def _configure_cmake(self):
...
             cmake.definitions["CURL_USE_WOLFSSL"] = self.options.with_ssl == "wolfssl"
-        elif tools.Version(self.version) >= "7.70.0":
+        elif tools.scm.Version(self.version) >= "7.70.0":
```
Comment (ericLemanissier) [inline recipes/libcurl/all/conanfile.py:547]:
```suggestion
        elif Version(self.version) >= "7.70.0":
```

File: recipes/libcurl/all/conanfile.py [hunk 16]
```diff
@@ -532,26 +532,26 @@ def _configure_cmake(self):
...
             cmake.definitions["CMAKE_USE_WOLFSSL"] = self.options.with_ssl == "wolfssl"
         cmake.definitions["USE_NGHTTP2"] = self.options.with_nghttp2
         cmake.definitions["CURL_ZLIB"] = self.options.with_zlib
         cmake.definitions["CURL_BROTLI"] = self.options.with_brotli
         if self._has_zstd_option:
             cmake.definitions["CURL_ZSTD"] = self.options.with_zstd
-        if tools.Version(self.version) >= "7.81.0":
+        if tools.scm.Version(self.version) >= "7.81.0":
```
Comment (ericLemanissier) [inline recipes/libcurl/all/conanfile.py:554]:
```suggestion
        if Version(self.version) >= "7.81.0":
```

File: recipes/libcurl/all/conanfile.py [hunk 17]
```diff
@@ -560,15 +560,15 @@ def _configure_cmake(self):
             cmake.definitions["ENABLE_THREADED_RESOLVER"] = self.options.with_threaded_resolver
         cmake.definitions["CURL_DISABLE_PROXY"] = not self.options.with_proxy
         cmake.definitions["USE_LIBRTMP"] = self.options.with_librtmp
-        if tools.Version(self.version) >= "7.75.0":
+        if tools.scm.Version(self.version) >= "7.75.0":
```
Comment (ericLemanissier) [inline recipes/libcurl/all/conanfile.py:563]:
```suggestion
        if Version(self.version) >= "7.75.0":
```

File: recipes/libcurl/all/conanfile.py [hunk 18]
```diff
@@ -560,15 +560,15 @@ def _configure_cmake(self):
...
             cmake.definitions["USE_LIBIDN2"] = self.options.with_libidn
         cmake.definitions["CURL_DISABLE_RTSP"] = not self.options.with_rtsp
         cmake.definitions["CURL_DISABLE_CRYPTO_AUTH"] = not self.options.with_crypto_auth
         cmake.definitions["CURL_DISABLE_VERBOSE_STRINGS"] = not self.options.with_verbose_strings
 
         # Also disables NTLM_WB if set to false
         if not self.options.with_ntlm:
-            if tools.Version(self.version) <= "7.77.0":
+            if tools.scm.Version(self.version) <= "7.77.0":
```
Comment (ericLemanissier) [inline recipes/libcurl/all/conanfile.py:571]:
```suggestion
            if Version(self.version) <= "7.77.0":
```

File: recipes/libcurl/all/conanfile.py [hunk 19]
```diff
@@ -639,7 +639,7 @@ def package_info(self):
             if self.options.with_ssl == "schannel":
                 self.cpp_info.components["curl"].system_libs.append("crypt32")
         elif tools.is_apple_os(self.settings.os):
-            if tools.Version(self.version) >= "7.77.0":
+            if tools.scm.Version(self.version) >= "7.77.0":
```
Comment (ericLemanissier) [inline recipes/libcurl/all/conanfile.py:642]:
```suggestion
            if Version(self.version) >= "7.77.0":
```

**Comment by prince-chrismc** [comment]
Please do not force push 🙏 GitHub forces us to restart the review which is not fun!

**Comment by prince-chrismc** [comment]
Please follow eric's suggestions 🙏

**Pull request closed without merging**